src/track/beats.h

```diff
@@ -24,29 +24,69 @@ typedef std::shared_ptr<const Beats> BeatsPointer;
 /// A beat marker is denotes the border of a tempo section inside a track.
 class BeatMarker {
   public:
-    BeatMarker(mixxx::audio::FramePos position, int beatsTillNextMarker)
-            : m_position(position), m_beatsTillNextMarker(beatsTillNextMarker) {
+    BeatMarker(mixxx::audio::FramePos position, mixxx::audio::FrameDiff_t beatLength, int barLength)
+            : m_position(position),
+              m_beatLength(beatLength),
+              m_beatsPerBar(barLength) {
         DEBUG_ASSERT(m_position.isValid());
         DEBUG_ASSERT(!m_position.isFractional());
-        DEBUG_ASSERT(m_beatsTillNextMarker > 0);
+        DEBUG_ASSERT(m_beatLength > 0);
+        DEBUG_ASSERT(m_beatsPerBar >= 2 && m_beatsPerBar <= 32);
```

Comment:
Well, its probably enough, but I wouldn't know what advantage a lower limit would have. So its difficult to make a decision since I don't know what the tradeoff is (if there is one at all)'